src/framework/accessors/player.ts

```diff
@@ -40,15 +49,17 @@ export function *getOpponents(
 export const hasEmptyPatrolSlot = (
   P: PlayerState | null,
 ): boolean => {
-  if (P == null) {
-    return false;
-  }
-  for (const slot of PATROL_SLOTS) {
-    if (P.patrol[slot] == null) {
-      return true;
-    }
+  return getFirstEmptyPatrolSlot(P) != null;
+};
+
+const getFirstEmptyPatrolSlot = (
+  P: PlayerState | null,
+): PatrolSlot | null => {
+  // eslint-disable-next-line no-unreachable-loop
+  for (const slot of getEmptyPatrolSlots(P)) {
```

Comment:
This looks like a good candidate for `return getEmptyPatrolSlots(P).find(() => true) | null;`

Reply:
:zap: